boot/freeldr/bootsect/ext.S

```diff
@@ -0,0 +1,490 @@
+/*
+ * PROJECT:     FreeLoader
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     EXTFS volume boot sector
+ * COPYRIGHT:   Copyright 2002-2003 Brian Palmer <brianp@sginet.com>
+ *              Copyright 2024 Daniel Victor <ilauncherdeveloper@gmail.com>
+ */
+
+#include <asm.inc>
+#include <freeldr/include/arch/pc/x86common.h>
+
+#define BP_REL(x) ss:[bp+x-offset start]
+
+#define EXT_INODE_BLOCKS                12
+#define EXT_INODE_INDIRECT_BLOCKS       3
+#define EXT_ROOT_INO			        2
+#define EXT_BLOCK_SIZE	                1024
+#define EXT_INODE_ADDRESS		        HEX(9000)
+#define EXT_DIRECTORY_ADDRESS	        HEX(A000)
+#define EXT_BLOCK_ADDRESS	            HEX(B000)
+#define EXT_BLOCK2_ADDRESS	            HEX(C000)
+#define EXT_INODE_TYPE_MASK             HEX(F000)
+#define EXT_INODE_TYPE_REGULAR          HEX(8000)
+
+.code16
+
+start:
+    jmp main
+.org 3
+
+BootDrive:
+    .byte HEX(80)
+ExtVolumeStartSector:
+    .long 263088				// Start sector of the ext2 volume
+ExtBlockSize:
+    .long 2						// Block size in sectors
+ExtBlockSizeInBytes:
+    .long 1024					// Block size in bytes
+ExtPointersPerBlock:
+    .long 256					// Number of block pointers that can be contained in one block
+ExtGroupDescSize:
+    .long 32 					// Size of Group Descriptor
+ExtGroupDescPerBlock:
+    .long 32					// Number of group descriptors per block
+ExtFirstDataBlock:
+    .long 2						// First data block (2 for 1024-byte blocks, 1 for bigger sizes)
+ExtInodeSize:
+    .long 128					// Size of Inode
+ExtInodesPerGroup:
+    .long 2048					// Number of inodes per group
+ExtInodesPerBlock:
+    .long 8						// Number of inodes per block
```

Comment:
Are all these variables above those that can (and will?) be adjusted by the bootsector installer at runtime?

Reply:
Yep